src/coreclr/vm/virtualcallstub.cpp

```diff
@@ -1114,27 +1268,61 @@ BYTE *VirtualCallStubManager::GenerateStubIndirection(PCODE target, BOOL fUseRec
     // Allocate from loader heap
     if (!ret)
     {
+        size_t alignment;
+        INTERFACE_DISPATCH_CACHED_OR_VSD(alignment = sizeof(TADDR) * 2, alignment = sizeof(TADDR));
+
         // Free list is empty, allocate a block of indcells from indcell_heap and insert it into the free list.
-        BYTE ** pBlock = (BYTE **) (void *) indcell_heap->AllocMem(S_SIZE_T(cellsPerBlock) * S_SIZE_T(sizeof(BYTE *)));
+        size_t cellsAllocationSize = cellsPerBlock * sizeOfIndCell;
+        size_t allocationSize = cellsAllocationSize;
+
+#ifdef FEATURE_CACHED_INTERFACE_DISPATCH
+        if (m_loaderAllocator->IsCollectible() && UseCachedInterfaceDispatch())
+        {
+            allocationSize += sizeof(CachedIndirectionCellBlockListNode);
+        }
+#endif // FEATURE_CACHED_INTERFACE_DISPATCH
+        BYTE ** pBlock = (BYTE **) (void *) indcell_heap->AllocAlignedMem(cellsPerBlock * sizeOfIndCell, alignment);
+
+#ifdef FEATURE_CACHED_INTERFACE_DISPATCH
```

Comment:
I see us assuming that pBlock is non-NULL in multiple places here, but the allocation could theoretically fail. Do we need a null check somewhere here?

Reply:
I believe the AllocAlignedMem throws OutOfMemoryException in case the allocation fails.

Reply:
Yes. Thank you @janvorli